### PR TITLE
PP-6362 - Add screen width to parameters

### PR DESCRIPTION
**Pull request by nimalank7**
Description:
- This PR adds screen width as a parameter for ePDQ 3DS2

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java [hunk 1]
```diff
@@ -35,11 +37,21 @@ public EpdqPayloadDefinitionForNew3ds2Order(String frontendUrl, SupportedLanguag
...
     }
 
+    private String getBrowserScreenWidth(EpdqTemplateData templateData) {
+        return templateData.getAuthCardDetails().getJsScreenWidth()
+                .filter(screenWidth -> NUMBER_FROM_0_TO_999999.matcher(screenWidth).matches())
+                .map(Integer::parseInt)
+                .filter(screenWidth -> screenWidth >= 0 && screenWidth <= 999999)
+                .map(screenWidth -> Integer.toString(screenWidth))
+                .orElse(DEFAULT_BROWSER_SCREEN_WIDTH);
+    }
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java:53]: We could try to make this share common code with `getBrowserScreenHeight(…)` but given I think that would probably lose clarity, so I’m fine with the way it is here.

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java [hunk 2]
```diff
@@ -48,7 +60,7 @@ private String getBrowserScreenHeight(EpdqTemplateData templateData) {
                 .map(screenHeight -> Integer.toString(screenHeight))
                 .orElse(DEFAULT_BROWSER_SCREEN_HEIGHT);
     }
-
+
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java]: Don’t think we want this indentation.

File: src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java [hunk 3]
```diff
@@ -104,6 +106,25 @@ public void should_include_browserScreenHeight(String screenHeight) {
         List<NameValuePair> result = epdqPayloadDefinitionFor3ds2NewOrder.extract(epdqTemplateData);
         assertThat(result, is(aParameterBuilder().build()));
     }
+
+    @Test
+    @Parameters({"0", "999999", "100"})
+    public void should_include_browserScreenWidth(String screenHeight) {
+        authCardDetails.setJsScreenHeight(screenHeight);
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java]: A little too eager with the copy/paste here!

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Looks good except for the copypasta.

File: src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java (hunk 3 above)
Comment (nimalank7) [reply src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java]: Ha - fixed

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java (hunk 1 above)
Comment (nimalank7) [reply src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java:53]: Yep

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java (hunk 2 above)
Comment (nimalank7) [reply src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java]: Done

**Review by alexbishop1 — APPROVED** [review]

**Pull request merged**